NineChronicles.Headless.Tests/GraphTypes/StandaloneQueryTest.cs

```diff
@@ -820,6 +820,104 @@ public async Task StateQueryShopState(ItemType itemType)
+                            level
+                            combatPoint
+                            armorId
+                            active
+                            dailyChallengeCount
+                            score
+                        }
+                    }
+                }
+            }";
+            var queryResult = await ExecuteQueryAsync(query);
+            Assert.Equal(
+                new Dictionary<string, object>
+                {
+                    ["stateQuery"] =new Dictionary<string, object>
+                    {
+                        ["weeklyArena"] = new Dictionary<string, object>
+                        {
+                            ["address"] = WeeklyArenaState.DeriveAddress(0).ToString(),
+                            ["ended"] = false,
+                            ["orderedArenaInfos"] = new List(),
+                        },
+                    },
+                },
+                queryResult.Data
+            );
+
+            var action = new SetAvatarState();
+            StandaloneContextFx.BlockChain.MakeTransaction(
+                userPrivateKey,
+                new PolymorphicAction<ActionBase>[] { action }
+            );
+            await blockChain.MineBlock(userAddress);
+
+            queryResult = await ExecuteQueryAsync(query);
+            Assert.Equal(
+                new Dictionary<string, object>
+                {
+                    ["stateQuery"] =new Dictionary<string, object>
+                    {
+                        ["weeklyArena"] = new Dictionary<string, object>
+                        {
+                            ["address"] = WeeklyArenaState.DeriveAddress(0).ToString(),
+                            ["ended"] = false,
+                            ["orderedArenaInfos"] = new List<object>
+                            {
+                                new Dictionary<string, object>
+                                {
+                                    ["agentAddress"] = userAddress.ToString(),
+                                    ["avatarAddress"] = avatarAddress.ToString(),
+                                    ["arenaRecord"] = new Dictionary<string, object>
+                                    {
+                                        ["win"] = 0,
+                                        ["lose"] = 0,
+                                        ["draw"] = 0,
+                                    },
+                                    ["level"] = 1,
+                                    ["combatPoint"] = 1142,
+                                    ["armorId"] = 10200000,
+                                    ["active"] = false,
+                                    ["dailyChallengeCount"] = 5,
+                                    ["score"] = 1000,
+                                }
+                            },
+                        },
+                    },
+                },
+                queryResult.Data
+            );
+        }
```

Comment:
I want to suggest to test in more smaller scope. It seems to test `SetAvatarState` action also. So I suggest to make a new test class like `WeeklyArenaStateTest`. You can see `CollectionMapTypeTest` as an example.

https://github.com/planetarium/NineChronicles.Headless/blob/6e4b0892624ec007fced271b9785047257bae0d0/NineChronicles.Headless.Tests/GraphTypes/States/Models/CollectionMapTypeTest.cs#L23-L25

Please leave opinions if you have! Or if you agreed with my suggestion but you are hard to acheive it, please leave any question also, don't feel hard 😀

Reply:
My intention was check cases when `OrderdArenaInfos` is empty and has a value. i'll try simplify the test by using `Theory`.